diff -up pyside-setup-opensource-src-5.15.2/build_scripts/setup_runner.py.omv~ pyside-setup-opensource-src-5.15.2/build_scripts/setup_runner.py
--- pyside-setup-opensource-src-5.15.2/build_scripts/setup_runner.py.omv~	2020-11-24 00:56:29.514811673 +0100
+++ pyside-setup-opensource-src-5.15.2/build_scripts/setup_runner.py	2020-11-24 00:57:03.778234854 +0100
@@ -138,7 +138,7 @@ class SetupRunner(object):
             # Reuse the shiboken build for the generator package instead
             # of rebuilding it again.
             self.add_setup_internal_invocation(config.shiboken_generator_option_name,
-                                               reuse_build=True)
+                                               reuse_build=False)
 
             self.add_setup_internal_invocation(config.pyside_option_name)
 
